packages/scanner/src/lib/eventAggregator.ts

```diff
@@ -0,0 +1,39 @@
+import { EventEmitter } from 'events';
+
+export type PendingEvent<E> = {
+  emitter: EventEmitter;
+  event: string;
+  args: E[];
```

Comment:
This isn't the right typing — it's limited to events that have all arguments of the same type. Either change it to `args: E` (and `E` could be a `[ScanEvent]` single-element tuple) or to `arg: E` (to explicitly limit it to single-argument events).